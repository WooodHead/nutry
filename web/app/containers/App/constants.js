/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const DEFAULT_LOCALE = 'en';
/*
 *
 * LandingPage constants
 *
 */
const prefix = (name) => `app/App/${name}`;
export const DEFAULT_ACTION = prefix('DEFAULT_ACTION');
export const LOGIN_FAILURE = prefix('LOGIN_FAILURE');
export const LOGIN_SUCCESS = prefix('LOGIN_SUCCESS');
export const LOGIN_CLICKED = prefix('LOGIN_CLICKED');
