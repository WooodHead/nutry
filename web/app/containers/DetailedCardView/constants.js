/*
 *
 * DetailedCardView constants
 *
 */

export const DEFAULT_ACTION = 'app/DetailedCardView/DEFAULT_ACTION';
