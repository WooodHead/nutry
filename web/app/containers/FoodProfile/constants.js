/*
 *
 * FoodProfile constants
 *
 */
import { Map, List } from 'immutable';
const prefix = (name) => `app/FoodProfile/${name}`;

export const ON_BACK = prefix('ON_BACK');

export const GET_PROFILE = prefix('GET_PROFILE');
export const GET_PROFILE_SUCCESS = prefix('GET_PROFILE_SUCCESS');
export const GET_PROFILE_FAILURE = prefix('GET_PROFILE_FAILURE');

export const TAB_CHANGED = prefix('TAB_CHANGED');
export const PORTION_CHANGED = prefix('PORTION_CHANGED');
export const NUTRIENT_SELECTED = prefix('NUTRIENT_SELECTED');
export const AGE_GROUP_CHANGED = prefix('AGE_GROUP_CHANGED');

export const HEADER = 'HEADER';
export const PARENT_ROW = 'PARENTR_ROW';
export const PARENT_NONAME_ROW = 'PARENT_NONAME_ROW';
export const CHILD_ROW = 'CHILD_ROW';
export const CHILD2_ROW = 'CHILD2_ROW';
export const SUMMARY_IDS = [
  {
    prefix: 'ENERC',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'ENERC_KCAL',
    type: PARENT_NONAME_ROW,
    selected: false,
  },
  {
    prefix: 'CHOCDF',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'SUGAR',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'FIBTG',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'PROCNT',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'FAT',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'NA',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'K',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'VITC',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'VITD',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'WATER',
    type: PARENT_ROW,
    selected: false,
  },
];
export const DETAILED_IDS = [
  // energy
  'ENERC', 'ENERC_KCAL',
  // fats
  'FAT', 'FASAT', 'FAMS', 'FAPU', 'FATRN', 'LCW3TOTAL',
  // carbohydrates
  'CHOCDF', 'FIBTG', 'SUGAR', 'GLUS', 'MALS', 'FRUS', 'LACS', 'GALS',
  // proteins
  'PROCNT', 'TRP', 'THR', 'ILE', 'LEU', 'LYS', 'MET', 'CYS', 'PHE', 'TYR', 'VAL',
  'ARG', 'HISTN', 'ALA', 'ASP', 'GLU', 'PRO', 'SER', 'HYP',
  // minerals
  'CA', 'FE', 'MG', 'P', 'K', 'NA', 'ZN', 'CU', 'MN', 'FLD',
  // vitamins
  'VITC', 'VITA_IU', 'RETOL', 'VITA_RAE', 'CARTA', 'CARTB', 'CRYPX', 'LYCPN',
  'LUT+ZEA', 'VITD', 'TOCPHA', 'TOCPHB', 'TOCPHD', 'VITK', 'THIA', 'NIA', 'SE',
  'FOL', 'FOLFD', 'FOLAC', 'FOLDFE', 'VITB6A', 'CHOLN', 'BETN',
  // basics
  'WATER',
];
export const DETAILED_SECTIONS = {
  energy: {
    headerName: 'Energy',
    items: [
      {
        prefix: 'ENERC',
        type: PARENT_ROW,
        selected: false,
      }, {
        prefix: 'ENERC_KCAL',
        type: PARENT_NONAME_ROW,
        selected: false,
      }],
  },
  fat: {
    headerName: 'Fat & Fatty Acids',
    items: [
      {
        prefix: 'FAT',
        type: PARENT_ROW,
        selected: false,
      }, {
        prefix: 'FASAT',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'FAMS',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'FAPU',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'FATRN',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'LCW3TOTAL',
        type: PARENT_ROW,
        selected: false,
      }],
  },
  carbohydrates: {
    headerName: 'Carbohydrates',
    items: [
      {
        prefix: 'CHOCDF',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FIBTG',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'SUGAR',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'GLUS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'SUCS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'MALS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'FRUS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'LACS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'GALS',
        type: CHILD2_ROW,
        selected: false,
      },
    ],
  },
  protein: {
    headerName: 'Proteins',
    items: [
      {
        prefix: 'PROCNT',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'TRP',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'THR',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ILE',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LEU',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LYS',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'MET',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CYS',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'PHE',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'TYR',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VAL',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ARG',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'HISTN',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ALA',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ASP',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'GLU',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'PRO',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'SER',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'HYP',
        type: CHILD_ROW,
        selected: false,
      },
    ],
  },
  minerals: {
    headerName: 'Minerals',
    items: [
      {
        prefix: 'CA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'MG',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'P',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'K',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'NA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'ZN',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'CU',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'MN',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'SE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FLD',
        type: PARENT_ROW,
        selected: false,
      },
    ],
  },
  vitamins: {
    headerName: 'Vitamins',
    items: [
      {
        prefix: 'VITC',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'VITA_IU',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'RETOL',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VITA_RAE',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CARTA',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CARTB',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CRYPX',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LYCPN',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LUT+ZEA',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VITD',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'TOCPHA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'TOCPHB',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'TOCPHD',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VITK',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'THIA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'NIA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'SE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOL',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOLFD',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOLAC',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOLDFE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'VITB6A',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'CHOLN',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'BETN',
        type: PARENT_ROW,
        selected: false,
      },
    ],
  },
};
export const FILTERS = {
  summary: [
    'CHOCDF',
    'SUGAR',
    'FIBTG',
    'PROCNT',
    'FAT',
    'WATER',
  ],
  detailed: [
    'CHOCDF',
    'SUGAR',
    'FIBTG',
    'PROCNT',
    'FAT',
    'WATER',
  ],
};
export const INITIAL_STATE = {
  loading: true,
  serialNumber: false,
  profileHeader: {},
  error: false,
  source: false,
  nutrients: {
    byId: {},
    bySection: Map(DETAILED_SECTIONS),
    bySummaryIds: SUMMARY_IDS,
    byDetailedIds: DETAILED_IDS,
    bySummaryPie: List([]),
    byDetailedPie: List([]),
  },
  tabSelected: 'summary',
  idSelected: false,
  nutrientSelected: '',
  portionSelected: { amt: 1,
    className: 'per100g',
    g: 100,
    label: 'per 100g',
    unit: 'per 100g',
    value: 100,
  },
  portionsAvailable: [],
  ageGroupSelected: { value: 'AM19', label: 'Adult Male (19-30)', className: 'am-19' },
};
