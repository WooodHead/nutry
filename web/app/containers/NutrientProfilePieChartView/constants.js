/*
 *
 * NutrientProfilePieChartView constants
 *
 */

export const DEFAULT_ACTION = 'app/NutrientProfilePieChartView/DEFAULT_ACTION';
