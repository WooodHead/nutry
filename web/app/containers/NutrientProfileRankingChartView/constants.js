/*
 *
 * NutrientProfileRankingChartView constants
 *
 */
const prefix = (name) => `app/NutrientProfileRankingChartView/${name}`;
export const DEFAULT_ACTION = prefix('DEFAULT_ACTION');

export const GET_NUTRIENT_RANKING = prefix('GET_NUTRIENT_RANKING');
export const GET_NUTRIENT_RANKING_SUCCESS = prefix('GET_NUTRIENT_RANKING_SUCCESS');
export const GET_NUTRIENT_RANKING_FAILURE = prefix('GET_NUTRIENT_RANKING_FAILURE');
