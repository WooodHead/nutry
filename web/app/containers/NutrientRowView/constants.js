/*
 *
 * NutrientRowView constants
 *
 */

export const DEFAULT_ACTION = 'app/NutrientRowView/DEFAULT_ACTION';
