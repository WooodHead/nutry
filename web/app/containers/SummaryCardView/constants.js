/*
 *
 * SummaryCardView constants
 *
 */

export const DEFAULT_ACTION = 'app/SummaryCardView/DEFAULT_ACTION';
