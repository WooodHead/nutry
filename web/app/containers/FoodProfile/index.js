/**
 *
 * FoodProfile
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { Helmet } from 'react-helmet';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { Row, Col } from 'antd';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import NutrientDisplay from 'components/NutrientDisplay';
import FoodProfileTitle from 'components/FoodProfileTitle';
import FoodProfileToolBar from 'components/FoodProfileToolBar';
import NutrientProfilePieChartView from 'containers/NutrientProfilePieChartView';
import NutrientProfileRankingChartView from 'containers/NutrientProfileRankingChartView';
import {
  loadProfile,
  tabChanged,
  portionChanged,
  ageGroupChanged } from './actions';
import {
  makeSelectProfileHeader,
  makeSelectProfileLoading,
  makeSelectTabSelected,
  makeSelectAllNutrients,
  makeSelectNutrientsBySummaryIds,
  makeSelectNutrientsBySections,
  makeSelectNutrientSelected,
  makeSelectAgeGroup,
  makeSelectPortion,
  makeSelectPortions,
 } from './selectors';
import reducer from './reducer';
import saga from './saga';
import FoodProfileWrapper from './FoodProfileWrapper';

export class FoodProfile extends React.Component { // eslint-disable-line react/prefer-stateless-function
  // componentWillMount() {
  //   // TODO: Add a prefix on results mouse-down to kick of the fetch extra early
  // }
  componentDidMount() {
    if (this.props.loading) {
      this.props.onLoadProfile(this.props.match ? this.props.match.params.profileId : this.props.profileHeader.SN);
    }
  }
  componentDidUpdate() {
    if (this.props.loading) {
      this.props.onLoadProfile(this.props.match ? this.props.match.params.profileId : this.props.profileHeader.SN);
    }
  }

  render() {
    const profileBackgroundStyle = {
      width: '100%',
      height: '100%',
      backgroundColor: '#DBD8D8',
    };
    const { loading,
      portions,
      profileHeader,
      portionSelected,
      ageGroupSelected } = this.props;
    const SN = this.props.match ? this.props.match.params.profileId : this.props.profileHeader.SN;
    const { onTabChange,
      onPortionChanged,
      onAgeGroupChanged,
      onLoadNewProfile,
     } = this.props;
    const profileTitleProps = {
      loading,
      profileHeader,
    };
    const nutrientDisplayProps = {
      loading,
      portionSelected,
      onTabChange,
    };

    const foodProfileToolbarProps = {
      loading,
      portions,
      portionSelected,
      ageGroupSelected,
      onAgeGroupChanged,
      onPortionChanged,
    };

    return (
      <FoodProfileWrapper>
        <div className="profile-background" style={profileBackgroundStyle}>
          <Helmet>
            <title>FoodProfile</title>
            <meta name="description" content="Description of FoodProfile" />
          </Helmet>
          <Row gutter={{ xs: 0, sm: 0, md: 0, lg: 24, xl: 48 }}>
            <Col xs={2} sm={2} md={2} lg={1}>
            </Col>
            <Col xs={20} sm={20} md={20} lg={22} >
              <FoodProfileTitle {...profileTitleProps} />
            </Col>
            <Col xs={2} sm={2} md={2} lg={1}>
            </Col>
          </Row>
          <Row>
            <Col xs={2} sm={2} md={2} lg={1}>
            </Col>
            <Col xs={20} sm={20} md={20} lg={22} >
              <FoodProfileToolBar {...foodProfileToolbarProps} />
            </Col>
            <Col xs={2} sm={2} md={2} lg={1}>
            </Col>
          </Row>
          <Row gutter={{ xs: 48, sm: 48, md: 48, lg: 48, xl: 48 }}>
            <Col xs={2} sm={2} md={1} lg={1}>
            </Col>
            <Col xs={20} sm={20} md={12} lg={12} >
              <NutrientDisplay {...nutrientDisplayProps} />
            </Col>
            <Col xs={20} sm={20} md={10} lg={10}>
              <NutrientProfilePieChartView />
              <NutrientProfileRankingChartView onLoadNewProfile={onLoadNewProfile} id={SN} />
            </Col>
            <Col xs={2} sm={2} md={1} lg={1}>
            </Col>
          </Row>
        </div>
      </FoodProfileWrapper>
    );
  }
}

FoodProfile.propTypes = {
  loading: PropTypes.bool,
  profileHeader: PropTypes.object.isRequired,
  onLoadProfile: PropTypes.func,
  onTabChange: PropTypes.func,
  portions: PropTypes.array,
  portionSelected: PropTypes.object,
  onAgeGroupChanged: PropTypes.func,
  onPortionChanged: PropTypes.func,
  ageGroupSelected: PropTypes.object,
  onLoadNewProfile: PropTypes.func,
  match: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  loading: makeSelectProfileLoading(),
  profileHeader: makeSelectProfileHeader(),
  tabSelected: makeSelectTabSelected(),
  portions: makeSelectPortions(),
  nutrients: makeSelectAllNutrients(),
  nutrientSelected: makeSelectNutrientSelected(),
  nutrientsBySummaryIds: makeSelectNutrientsBySummaryIds(),
  nutrientsBySections: makeSelectNutrientsBySections(),
  nutrientFocused: makeSelectNutrientSelected(),
  portionSelected: makeSelectPortion(),
  ageGroupSelected: makeSelectAgeGroup(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
    onLoadProfile: (serialNumber) => dispatch(loadProfile(serialNumber)),
    onLoadNewProfile: (profileId) => dispatch(push(`/foodprofile/${profileId}`)),
    onTabChange: (tab) => dispatch(tabChanged(tab)),
    onPortionChanged: (portion) => dispatch(portionChanged(portion)),
    onAgeGroupChanged: (ageGroup) => dispatch(ageGroupChanged(ageGroup)),
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'foodProfile', reducer });
const withSaga = injectSaga({ key: 'foodProfile', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(FoodProfile);
