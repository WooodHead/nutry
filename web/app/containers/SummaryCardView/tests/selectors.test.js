import { fromJS } from 'immutable';
// import { selectDetailedCardViewDomain } from '../selectors';
import { SUMMARY_IDS } from 'containers/FoodProfile/constants';
import {
  makeSelectBySummaryIds,
} from '../selectors';

describe('Container [DetailedCardView] - selectors', () => {
  describe('makeSelectLocation', () => {
    it('should select the location', () => {
      const foodProfile = fromJS({
        nutrients: { bySummaryIds: SUMMARY_IDS },
      });
      const mockedState = fromJS({
        foodProfile,
      });
      expect(makeSelectBySummaryIds()(mockedState)).toMatchSnapshot();
    });
  });
});
