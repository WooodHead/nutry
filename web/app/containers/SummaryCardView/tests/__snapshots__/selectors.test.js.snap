// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [DetailedCardView] - selectors makeSelectLocation should select the location 1`] = `
Immutable.List [
  Immutable.Map {
    "prefix": "ENERC",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "ENERC_KCAL",
    "type": "PARENT_NONAME_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "CHOCDF",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "SUGAR",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "FIBTG",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "PROCNT",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "FAT",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "NA",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "K",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "VITC",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "VITD",
    "type": "PARENTR_ROW",
    "selected": false,
  },
  Immutable.Map {
    "prefix": "WATER",
    "type": "PARENTR_ROW",
    "selected": false,
  },
]
`;
